File: gluecodium/src/test/resources/smoke/attributes/output/android/com/example/smoke/AttributesWithComments.java
/*
 *
 */
package com.example.smoke;
import android.support.annotation.NonNull;
import com.example.NativeBase;
/**
 * <p>Class comment
 */
@OnClass
public final class AttributesWithComments extends NativeBase {
    /**
     * <p>Const comment
     */
    @OnConstInClass
    public static final boolean PI = false;
    public static final class SomeStruct {
        /**
         * <p>Field comment
         */
        @OnField
        @NonNull
        public String field;
        public SomeStruct() {
            this.field = "";
        }
    }
    /**
     * For internal use only.
     * @hidden
     * @param nativeHandle The SDK nativeHandle instance.
     * @param dummy The SDK dummy instance.
     */
    protected AttributesWithComments(final long nativeHandle, final Object dummy) {
        super(nativeHandle, new Disposer() {
            @Override
            public void disposeNative(long handle) {
                disposeNativeHandle(handle);
            }
        });
    }
    private static native void disposeNativeHandle(long nativeHandle);
    /**
     * <p>Function comment
     */
    @OnFunctionInClass
    public native void veryFun();
    /**
     * <p>Getter comment
     * @return <p>Property comment
     */
    @OnPropertyInClass
    @NonNull
    public native String getProp();
    /**
     * <p>Setter comment
     * @param value <p>Property comment
     */
    @OnPropertyInClass
    public native void setProp(@NonNull final String value);
}

File: gluecodium/src/test/resources/smoke/attributes/output/android-kotlin/com/example/smoke/AttributesWithComments.kt
/*

 *
 */

@file:JvmName("AttributesWithComments")

package com.example.smoke

import com.example.NativeBase

/**
 * Class comment
 */
@OnClass
class AttributesWithComments : NativeBase {

    class SomeStruct {
        /**
         * Field comment
         */
        @OnField
        @JvmField var field: String



        constructor() {
            this.field = ""
        }





    }



    /**
     * For internal use only.
     * @suppress
     * @param nativeHandle The handle to resources on C++ side.
     * @param tag Tag used by callers to avoid overload resolution problems.
     */
    protected constructor(nativeHandle: Long, @Suppress("UNUSED_PARAMETER") tag: Any?)
        : super(nativeHandle, { disposeNativeHandle(it) }) {}



    /**
     * Function comment
     */
    @OnFunctionInClass

    external fun veryFun() : Unit

    /**
     * Property comment
     */
    var prop: String
        external get
        external set




    companion object {
        /**
         * Const comment
         */
        @OnConstInClass
        @JvmField final val PI: Boolean = false
        @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)
    }
}

